### PR TITLE
Fix a bunch of typos

### PR DESCRIPTION
**Pull request by oxan**
# What does this implement/fix? 

Fix a bunch of typos.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/nextion/nextion.h [hunk 1]
```diff
@@ -546,7 +546,7 @@ class Nextion : public NextionBase, public PollingComponent, public uart::UARTDe
    * it.filled_cricle(25, 25, 10, "17013");
    * ```
    *
-   * Makes a filled circle at the x cordinates `25` and y coordinate `25` with a radius of `10` with a color of blue.
+   * Makes a filled circle at the x coordinates `25` and y coordinate `25` with a radius of `10` with a color of blue.
```
Comment (agners) [inline esphome/components/nextion/nextion.h]:
```suggestion
   * Makes a filled circle at the x coordinate `25` and y coordinate `25` with a radius of `10` with a color of blue.
```

File: esphome/components/nextion/nextion.h [hunk 2]
```diff
@@ -563,7 +563,7 @@ class Nextion : public NextionBase, public PollingComponent, public uart::UARTDe
    * it.filled_cricle(25, 25, 10, color);
    * ```
    *
-   * Makes a filled circle at the x cordinates `25` and y coordinate `25` with a radius of `10` with a color of blue.
+   * Makes a filled circle at the x coordinates `25` and y coordinate `25` with a radius of `10` with a color of blue.
```
Comment (agners) [inline esphome/components/nextion/nextion.h]:
```suggestion
   * Makes a filled circle at the x coordinate `25` and y coordinate `25` with a radius of `10` with a color of blue.
```

**Review by agners — APPROVED** [review]
Other than that, LGTM.

**Review by glmnet — APPROVED** [review]
Yup all comments and strings spell chekced 😜

File: esphome/components/st7789v/st7789v.h [hunk 3]
```diff
@@ -89,7 +89,7 @@ static const uint8_t ST7789_GATECTRL = 0xE4;   // Gate Control
 static const uint8_t ST7789_SPI2EN = 0xE7;     // SPI2 Enable
 static const uint8_t ST7789_PWCTRL2 = 0xE8;    // Power Control 2
 static const uint8_t ST7789_EQCTRL = 0xE9;     // Equalize time control
-static const uint8_t ST7789_PROMCTRL = 0xEC;   // Program Mode Contro
+static const uint8_t ST7789_PROMCTRL = 0xEC;   // Program Mode Control
 static const uint8_t ST7789_PROMEN = 0xFA;     // Program Mode Enabl
```
Comment (jesserockz) [inline esphome/components/st7789v/st7789v.h]:
```suggestion
static const uint8_t ST7789_PROMEN = 0xFA;     // Program Mode Enable
```

Comment (jesserockz) [reply esphome/components/st7789v/st7789v.h]: You missed one =)

File: esphome/components/st7789v/st7789v.h [hunk 4]
```diff
@@ -59,11 +59,11 @@ static const uint8_t ST7789_RDID1 = 0xDA;      // Read ID1
...
 static const uint8_t ST7789_RAMCTRL = 0xB0;    // RAM Control
-static const uint8_t ST7789_RGBCTRL = 0xB1;    // RGB Interface Contro
+static const uint8_t ST7789_RGBCTRL = 0xB1;    // RGB Interface Control
 static const uint8_t ST7789_PORCTRL = 0xB2;    // Porch Setting
 static const uint8_t ST7789_FRCTRL1 = 0xB3;    // Frame Rate Control 1 (In partial mode/ idle colors)
-static const uint8_t ST7789_PARCTRL = 0xB5;    // Partial mode Contro
-static const uint8_t ST7789_GCTRL = 0xB7;      // Gate Contro
+static const uint8_t ST7789_PARCTRL = 0xB5;    // Partial mode Control
+static const uint8_t ST7789_GCTRL = 0xB7;      // Gate Control
 static const uint8_t ST7789_GTADJ = 0xB8;      // Gate On Timing Adjustmen
```
Comment (jesserockz) [inline esphome/components/st7789v/st7789v.h]: And another here 
```suggestion
static const uint8_t ST7789_GTADJ = 0xB8;      // Gate On Timing Adjustment
```

Comment (jesserockz) [reply esphome/components/st7789v/st7789v.h]: Also a couple more above that GH wont let me suggest because its out of range of the PR

Comment (oxan) [reply esphome/components/st7789v/st7789v.h]: Okay, fixed a bunch more here as well.

File: esphome/components/aht10/aht10.cpp [hunk 5]
```diff
@@ -23,7 +23,7 @@ static const uint8_t AHT10_CALIBRATE_CMD[] = {0xE1};
 static const uint8_t AHT10_MEASURE_CMD[] = {0xAC, 0x33, 0x00};
 static const uint8_t AHT10_DEFAULT_DELAY = 5;    // ms, for calibration and temperature measurement
 static const uint8_t AHT10_HUMIDITY_DELAY = 30;  // ms
-static const uint8_t AHT10_ATTEMPS = 3;          // safety margin, normally 3 attemps are enough: 3*30=90ms
+static const uint8_t AHT10_ATTEMPS = 3;          // safety margin, normally 3 attempts are enough: 3*30=90ms
```
Comment (OttoWinter) [inline esphome/components/aht10/aht10.cpp]:
```suggestion
static const uint8_t AHT10_ATTEMPTS = 3;          // safety margin, normally 3 attempts are enough: 3*30=90ms
```

The comment is fixed, but not the name :stuck_out_tongue_winking_eye:

File: esphome/components/aht10/aht10.cpp [hunk 6]
```diff
@@ -59,7 +59,7 @@ void AHT10Component::update() {
   if (this->humidity_sensor_ != nullptr)
     delay = AHT10_HUMIDITY_DELAY;
   for (int i = 0; i < AHT10_ATTEMPS; ++i) {
```
Comment (OttoWinter) [inline esphome/components/aht10/aht10.cpp]:
```suggestion
  for (int i = 0; i < AHT10_ATTEMPTS; ++i) {
```

**Review by OttoWinter — APPROVED** [review]

**Review by jesserockz — APPROVED** [review]

File: esphome/components/aht10/aht10.cpp [hunk 7]
```diff
@@ -23,7 +23,7 @@ static const uint8_t AHT10_CALIBRATE_CMD[] = {0xE1};
 static const uint8_t AHT10_MEASURE_CMD[] = {0xAC, 0x33, 0x00};
 static const uint8_t AHT10_DEFAULT_DELAY = 5;    // ms, for calibration and temperature measurement
 static const uint8_t AHT10_HUMIDITY_DELAY = 30;  // ms
-static const uint8_t AHT10_ATTEMPS = 3;          // safety margin, normally 3 attemps are enough: 3*30=90ms
+static const uint8_t AHT10_ATTEMPTS = 3;          // safety margin, normally 3 attempts are enough: 3*30=90ms
```
Comment (OttoWinter) [inline esphome/components/aht10/aht10.cpp]:
```suggestion
static const uint8_t AHT10_ATTEMPTS = 3;         // safety margin, normally 3 attempts are enough: 3*30=90ms
```

**Pull request merged**